### PR TITLE
[feature #2793] Re-natal latest config.cljs

### LINKED ISSUE #2793: Rely on latest re-natal config.cljs
### User Story

As a developer, I don't want to have files part of `git` automatically modified by tools (e.g. `re-natal`)

### Description

Currently [android](https://github.com/status-im/status-react/blob/develop/env/dev/env/android/main.cljs) and [ios](https://github.com/status-im/status-react/blob/develop/env/dev/env/ios/main.cljs) main files are modified each time `re-natal` is run.
Rely on latest re-natal [changes](https://github.com/drapanjanas/re-natal/commit/5f6da94ae72f5a3fc162f79acc73843ff43925b0) that introduced [config.cljs](https://github.com/status-im/status-react/blob/develop/env/dev/env/config.cljs) to make sure those files won't be modified.
Make sure `re-frisk` configuration is also updated.

`config.cljs` will then be modified by `re-natal` but listed in `.gitignore`.

*Type*: Feature

### PR DESCRIPTION
**Pull request by t4sk**
Read Re-natal config from config.cljs

Fixes #2793 

### Summary:
[comment]: # update re-natal and avoid re-natal from updating files managed under git

- update re-natal to 0.8.0
- git ignore config.cljs
- derive re-frisk url from config.cljs
- add tests for env/dev/env/utils
- remove `lein re-frisk use-re-natal` from `Makefile` 

### Steps to test:
- start Status in dev mode
- open re-frisk (address shown in lein repl)
- `lein test-env-dev-utils`

status: ready

### CODE REVIEW

File: env/dev/env/android/main.cljs
Comment (jeluard) [inline env/dev/env/android/main.cljs]: We usually use full name as alias so `figwheel` was good here.

File: env/dev/env/ios/main.cljs
Comment (jeluard) [inline env/dev/env/ios/main.cljs]: Why the change to `start` ?

File: env/dev/env/util.cljs
Comment (jeluard) [inline env/dev/env/util.cljs]: We usually use `string` as alias.

File: env/dev/env/util.cljs
Comment (jeluard) [inline env/dev/env/util.cljs]: Can you add a comment detailing the expected `url` format? In case the `re-natal` format changes or we want to reuse it. Some tests would be even better !

**Review by jeluard — COMMENTED** [review]
Thanks! Looks great! Let's wait after christmas to have some people test locally.

File: env/dev/env/android/main.cljs
Comment (t4sk) [reply env/dev/env/android/main.cljs]: Thanks for your feedback.
When I ran `re-natal upgrade`, it automatically made changes to this file.
Will fix.

File: env/dev/env/ios/main.cljs
Comment (t4sk) [reply env/dev/env/ios/main.cljs]: Same reason as above.
Will fix

File: env/dev/env/util.cljs
Comment (t4sk) [reply env/dev/env/util.cljs]: Yup, will do

File: env/dev/env/ios/main.cljs
Comment (jeluard) [reply env/dev/env/ios/main.cljs]: It would be interesting to know what the difference is between both.

Comment (t4sk) [reply env/dev/env/ios/main.cljs]: according to here
https://github.com/bhauman/lein-figwheel/blob/master/support/src/figwheel/client.cljs#L479

`watch-and-reload` is a legacy interface which calls `start`

Comment (jeluard) [reply env/dev/env/ios/main.cljs]: Let's keep `start` then!

**Comment by flexsurfer** [comment]
hey @t4sk thanks for contribution, please take a look here https://github.com/flexsurfer/lein-re-frisk/blob/master/plugin/src/leiningen/re_frisk.clj#L16, it seems we need new version for lein-re-frisk, because we don't need to replace host anymore?

**Comment by flexsurfer** [comment]
thanks for contribution @t4sk , looks like we don't need to call `re-frisk use-re-natal` anymore

File: env/dev/env/ios/main.cljs
Comment (jeluard) [inline env/dev/env/ios/main.cljs]: Why the extra line?

**Review by jeluard — APPROVED** [review]
Looks good to me!

File: .gitignore
Comment (yenda) [inline .gitignore]: don't add line at the end of the file

File: env/dev/env/android/main.cljs
Comment (yenda) [inline env/dev/env/android/main.cljs]: don't add linebreak after opening brackets

File: env/dev/env/android/main.cljs
Comment (yenda) [inline env/dev/env/android/main.cljs]: don't add linebreak after opening brackets

File: env/dev/env/ios/main.cljs
Comment (yenda) [inline env/dev/env/ios/main.cljs]: don't add linebreak after opening brackets

**Review by yenda — COMMENTED** [review]
Thanks for your contribution good job, I just requested minor formatting changes

File: env/dev/env/ios/main.cljs
Comment (t4sk) [reply env/dev/env/ios/main.cljs]: @yenda Thanks for your feedback. I will address theses issues late next week.

**Comment by t4sk** [comment]
Hi all, thanks for your feedback. I will address theses issues late next week.

**Comment by t4sk** [comment]
Thank you all for your patience and happy new year.

**Comment by t4sk** [comment]
@flexsurfer
Thanks for your feedback. I am quit not sure where I need to make the modifications. Can you please give me suggestions?

**Comment by jeluard** [comment]
@t4sk I understand @flexsurfer means you can remove `re-frisk use-re-natal` from [Makefile](https://github.com/status-im/status-react/blob/develop/Makefile#L34).

File: env/dev/env/ios/main.cljs
Comment (flexsurfer) [inline env/dev/env/ios/main.cljs:13]: why do we have this only for ios?

File: env/dev/env/ios/main.cljs
Comment (flexsurfer) [inline env/dev/env/ios/main.cljs:18]: just wondering why do we need this magic?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: env/dev/env/ios/main.cljs
Comment (t4sk) [reply env/dev/env/ios/main.cljs:13]: It was already included in [android](https://github.com/status-im/status-react/blob/develop/env/dev/env/android/main.cljs#L12)

File: env/dev/env/ios/main.cljs
Comment (t4sk) [reply env/dev/env/ios/main.cljs:18]: It was auto-inserted when re-natal was updated. I decided to keep it since it was already present in [android/main.cljs](https://github.com/status-im/status-react/blob/develop/env/dev/env/android/main.cljs#L13)

Comment (yenda) [reply env/dev/env/ios/main.cljs:18]: this is re-natal auto-generated assertions, I think we can remove them

**Comment by jeluard** [comment]
@t4sk Can you look at @flexsurfer comments?

**Comment by t4sk** [comment]
@jeluard 
Sorry, I didn't know how to make my comments visible. It was in "pending" . Thanks for mentioning, otherwise I would have been waiting for a feedback :P

**Review by flexsurfer — APPROVED** [review]

**Comment by oskarth** [comment]
@yenda @jeluard @flexsurfer can either of you test and sanity check this on Android and iOS as well as CI? If it seems OK, let's merge it. QA pipeline too full.

**Comment by jeluard** [comment]
@oskarth I tried locally, works fine.

**Comment by oskarth** [comment]
Thanks @jeluard, will check CI and then merge later today.

**Comment by oskarth** [comment]
https://jenkins.status.im/job/status-react/job/parametrized%20build/ #282 pending build before merge

EDIT: Nevermind, I see PR CI works now :)

**Pull request merged**
